libsql-sqlite3/src/vectordiskann.c

```diff
@@ -1428,12 +1428,12 @@ int diskAnnSearch(
     *pzErrMsg = sqlite3_mprintf("vector index(search): k must be a non-negative integer");
     return SQLITE_ERROR;
   }
-  if( pIndex->nVectorDims != pVector->dims ){
+  if( pVector->dims != pIndex->nVectorDims ){
```

Comment:
Is there some reason for this change? I am fine with slipping it through, but doesn't seem to actually do anything?

Reply:
I just wanted to make conditions consistent: left argument is vector, right is index

Reply:
And below there are few more similar checks